spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -194,10 +205,40 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
     Map<NamedReference, ColumnStatistics> colStatsMap = Collections.emptyMap();
     if (readConf.reportColumnStats() && cboEnabled) {
       colStatsMap = Maps.newHashMap();
+      Map<Integer, Long> ndvs = Maps.newHashMap();
+      Map<Integer, Long> nullCounts = Maps.newHashMap();
+      Map<Integer, Object> minValues = Maps.newHashMap();
+      Map<Integer, Object> maxValues = Maps.newHashMap();
       List<StatisticsFile> files = table.statisticsFiles();
       if (!files.isEmpty()) {
         List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
 
+        if (readConf.deriveStatsFromManifestSessionConf()
+            || readConf.deriveStatsFromManifestTableProperty()) {
+          Map<String, Map<Integer, Long>> distinctDataFilesNullCount = Maps.newHashMap();
+          Map<String, Map<Integer, ByteBuffer>> distinctDataFilesMin = Maps.newHashMap();
+          Map<String, Map<Integer, ByteBuffer>> distinctDataFilesMax = Maps.newHashMap();
+
+          // extract the distinct files which are part of the task planning
+          for (ScanTaskGroup<?> taskGrp : taskGroups()) {
+            for (ScanTask task : taskGrp.tasks()) {
+              if (task.isFileScanTask()) {
+                FileScanTask fileScanTask = task.asFileScanTask();
```

Comment:
Could we just make a set of files being used?

Stream.of(tasks).filter(fl.isFileScanTask).map(file).collectAsSet

Then we could just work on the set for all of our min and max functions

Reply:
Updated to collect fileScanTasks into a set, Thanks!